docs/clients/promtail/README.md

```diff
@@ -72,6 +73,22 @@ This endpoint returns Promtail metrics for Prometheus. See
 "[Operations > Observability](../../operations/observability.md)" to get a list
 of exported metrics.
 
+### `POST /push/<label pairs>`
+
+When the [HTTP Target](./scraping.md#http-target) is being used, a `/push`
+endpoint is enabled that allows for writing logs:
+
+```
+curl -XPOST "http://promtail/push/label1/value1/label2/value2" -d 'hello, world!'
```

Comment:
This format looks weird to me .. is it really required to do `label/value` as part of the path? Especially HTTP API's nowadays are more or less RESTful, and the path should represent the location of the resource, but not actual data.

As a user I'd rather expect a more traditional JSON api:

```http
POST /push
Content-Type: application/json

{
  "line": "foo",
  "labels": {
    "bar": "baz",
  },
  "timestamp": "1257894000"
}
```

Are there any downsides of such an approach I am missing?

Reply:
It's modeled after the Prometheus [pushgateway API](https://github.com/prometheus/pushgateway#url). I'm personally a fan of how simple it is, especially for just pushing data to Promtail directly over the command line. 

There's been some debate over whether we should use a traditional JSON payload instead. I'm not opposed to changing it, but I went with the current implementation first just out of my own preference. I feel that embedding the labels and timestamp as metadata in the URL path is slightly easier to work with on the client side than it would be if JSON was required. 
 
There's also been questions about making this _just_ a gRPC method and not expose it as an HTTP1 endpoint at all. A consensus hasn't been reached yet, so you might want to hold off on reviewing this further for a while.

Reply:
The way it's currently written would be very useful to me (HTTP1 and label in the url path), as I often don't control the clients.  Leveraging Istio and this PR I should be able to easily mirror any traffic to promtail with arbitrary labels and look at it in grafana.  Requiring grpc or a specific json payload format would make this much more difficult.

Reply:
I second this. Simple, efficient and Prometheus-like